app/src/main/java/com/kickstarter/ui/activities/DeepLinkActivity.java

```diff
@@ -0,0 +1,79 @@
+package com.kickstarter.ui.activities;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Parcelable;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+
+import com.kickstarter.libs.BaseActivity;
+import com.kickstarter.libs.RefTag;
+import com.kickstarter.libs.qualifiers.RequiresActivityViewModel;
+import com.kickstarter.libs.utils.ApplicationUtils;
+import com.kickstarter.ui.IntentKey;
+import com.kickstarter.viewmodels.DeepLinkViewModel;
+
+import java.util.List;
+
+import static com.kickstarter.libs.rx.transformers.Transformers.observeForUI;
+
+@RequiresActivityViewModel(DeepLinkViewModel.ViewModel.class)
+public class DeepLinkActivity extends BaseActivity<DeepLinkViewModel.ViewModel> {
+  @Override
+  protected void onCreate(@Nullable final Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    this.viewModel.outputs.startDiscoveryActivity()
+      .compose(bindToLifecycle())
+      .compose(observeForUI())
+      .subscribe(__ -> startDiscoveryActivity());
+
+    this.viewModel.outputs.startProjectActivity()
+      .compose(bindToLifecycle())
+      .compose(observeForUI())
+      .subscribe(this::startProjectActivity);
+
+    this.viewModel.outputs.startBrowser()
+      .compose(bindToLifecycle())
+      .compose(observeForUI())
+      .subscribe(this::startBrowser);
+
+    this.viewModel.outputs.requestPackageManager()
+      .compose(bindToLifecycle())
+      .compose(observeForUI())
+      .subscribe(this::inputPackageManager);
+  }
+
+  private void startDiscoveryActivity() {
+    ApplicationUtils.startNewDiscoveryActivity(this);
+    finish();
+  }
+
+  private void startProjectActivity(final String url) {
+    final Uri uri = Uri.parse(url);
+    final Intent projectIntent = new Intent(this, ProjectActivity.class)
+      .setData(uri);
+    final String ref = uri.getQueryParameter("ref");
+    if (ref != null) {
+      projectIntent.putExtra(IntentKey.REF_TAG, RefTag.from(ref));
+    }
+    startActivity(projectIntent);
+    finish();
+  }
+
+  private void startBrowser(final List<Intent> targetIntents) {
+    final Intent chooserIntent = Intent.createChooser(targetIntents.remove(0), "");
+    chooserIntent.putExtra(Intent.EXTRA_INITIAL_INTENTS,
+      targetIntents.toArray(new Parcelable[targetIntents.size()]));
+    startActivity(chooserIntent);
+
+    finish();
+  }
+
+  private void inputPackageManager(final String url) {
+    if (!TextUtils.isEmpty(url)) {
```

Comment:
could we add a `.filter(url -> !TextUtils.isEmpty(url))` on the output here so that we won't get empty values emitted?  https://github.com/kickstarter/android-oss/pull/171/files#diff-2f428d6bea56ac66d4ec1a8567629475R105

unless I'm missing something and we actually want those